### DISCUSSION TITLE
Styling of labels and panels

### DISCUSSION THREAD

**Discussion by philstopford** [picoe/Eto#1855] [Q&A]
I've been trying to make a stab at a dark WPF theme here : 

https://github.com/philstopford/Eto/tree/dark_mode_experiment

using the test project to check validity. I have some lingering questions, particularly in terms of how panels and labels get styled. I tried digging down through the inheritance and couldn't see the right approach to use.

**Comment by philstopford** [comment]
To add some context to the above : the big white areas below are where I'm still having issues locating the right approach : 

![image](https://user-images.githubusercontent.com/1983851/103701580-e363de00-4f6b-11eb-8ae2-f71f599b5f33.png)

**Comment by cwensley** [comment]
@philstopford sorry I don't have a ton of time to look into this, but to help point you in the right direction it could either be the Form or the Splitter that is causing the background to be white there.

Hope this helps!

**Comment by cwensley** [comment]
@philstopford I hope to devote some time to this soonish, and add support to Eto for themes.  See #1896

**Comment by philstopford** [comment]
@cwensley : just wondered if you'd been able to look at this yet :)

**Comment by cwensley** [comment]
@philstopford try adding this line to Startup.cs:

```
Style.Add<Eto.Forms.Window>(null, w => w.BackgroundColor = Eto.Drawing.Colors.Transparent);
```

Eto sets the background color of the window's content to `System.Windows.SystemColors.ControlBrush` explicitly so doing the above reverts that.

**Comment by philstopford** [comment]
@cwensley : that makes a huge difference. Thanks!

Remaining issues:

- Clicking on the Label Controls entry in the listbox, all fonts go green and stay that way.
- Kitchen Sink has a white background and shows a bunch of other not-dark controls.
- ...

**Comment by cwensley** [comment]
Interesting, after playing with this a bit more it didn't make sense to me that the ControlBrush wasn't already overridden by the style..  It's quite annoying that `System.Windows.SystemColors.*` doesn't actually change when using a theme that overrides their values..

However, if I change this (in WpfWindow.cs):

```
content.Background = sw.SystemColors.ControlBrush;;
```
to this:
```
content.SetResourceReference(swc.Panel.BackgroundProperty, sw.SystemColors.ControlBrushKey);
```

.. then it picks up the dark color automatically, and still gets the right color if no theme is set.

So, anywhere in Eto using a system color should be changed to do the above, which would make things work better with custom WPF themes.

**Comment by cwensley** [comment]
As for KitchenSink, it's the TabControl.. Your xaml is theming the TabControlHandler and TabPageHandler, which doesn't do anything as they are not a WPF control:

```
    <Style x:Key="{x:Type eto:TabControlHandler}"
       BasedOn="{StaticResource {x:Type TabControl}}"
       TargetType="{x:Type eto:TabControlHandler}"/>

    <Style x:Key="{x:Type eto:TabPageHandler}"
       BasedOn="{StaticResource {x:Type TabControl}}"
       TargetType="{x:Type eto:TabPageHandler}"/>
```

Changing it to this works:
```
    <Style x:Key="{x:Type eto:EtoTabControl}"
       BasedOn="{StaticResource {x:Type TabControl}}"
       TargetType="{x:Type eto:EtoTabControl}"/>
```

**Comment by cwensley** [comment]
The label section changes everything as it modifies the color of the existing brush of the Label.. which probably will need to be changed in Eto to always create a new brush.

**Comment by philstopford** [comment]
My XAML efforts are entirely throw-at-wall :D

**Comment by philstopford** [comment]
Switching to the PropertyGrid also causes an instant crash.
